galsim/lsst/lsst_wcs.py

```diff
@@ -0,0 +1,1153 @@
+
+
+    def getTanWcs(self):
+        """
+        Return a WCS which approximates the focal plane as perfectly flat
+        (i.e. it ignores optical distortions that the telescope may impose on the image)
+
+        The output is an instantiation of lsst.afw.image's TanWcs class
+        representing the WCS of the detector as if there were no optical
+        distortions imposed by the telescope.
+        """
+
+        xTanPixMin, xTanPixMax, \
+        yTanPixMin, yTanPixMax = self._getTanPixelBounds()
+
+
+        xPixList = []
+        yPixList = []
+        nameList = []
+
+        #dx and dy are set somewhat heuristically
+        #setting them eqal to 0.1(max-min) lead to errors
+        #on the order of 0.7 arcsec in the WCS
+
+        dx = 0.5*(xTanPixMax-xTanPixMin)
+        dy = 0.5*(yTanPixMax-yTanPixMin)
+        for xx in np.arange(xTanPixMin, xTanPixMax+0.5*dx, dx):
+            for yy in np.arange(yTanPixMin, yTanPixMax+0.5*dx, dx):
+                xPixList.append(xx)
+                yPixList.append(yy)
+                nameList.append(self._chip_name)
+
+        raList, decList = self._camera.raDecFromTanPixelCoords(np.array(xPixList),
+                                                               np.array(yPixList),
+                                                               nameList)
+
+        raPointing = self._camera._pointing.ra/galsim.radians
+        decPointing = self._camera._pointing.dec/galsim.radians
+
+        camera_point_list = self._camera._get_afw_pupil_coord_list_from_float(raPointing, decPointing)
+        crPix1, crPix2 = self._camera._tan_pixel_coord_from_point_and_name(camera_point_list, [self._chip_name])
+
+        lonList, latList = _nativeLonLatFromRaDec(raList, decList, raPointing, decPointing)
+
+        #convert from native longitude and latitude to intermediate world coordinates
+        #according to equations (12), (13), (54) and (55) of
+        #
+        #Calabretta and Greisen (2002), A&A 395, p. 1077
+        #
+        radiusList = 180.0/(np.tan(latList)*np.pi)
+        uList = radiusList*np.sin(lonList)
+        vList = -radiusList*np.cos(lonList)
+
+        delta_xList = xPixList - crPix1[0]
+        delta_yList = yPixList - crPix2[0]
+
+        bVector = np.array([
+                           (delta_xList*uList).sum(),
+                           (delta_yList*uList).sum(),
+                           (delta_xList*vList).sum(),
+                           (delta_yList*vList).sum()
+                           ])
+
+        offDiag = (delta_yList*delta_xList).sum()
+        xsq = np.power(delta_xList,2).sum()
+        ysq = np.power(delta_yList,2).sum()
+
+        aMatrix = np.array([
+                           [xsq, offDiag, 0.0, 0.0],
+                           [offDiag, ysq, 0.0, 0.0],
+                           [0.0, 0.0, xsq, offDiag],
+                           [0.0, 0.0, offDiag, ysq]
+                           ])
+
+        coeffs = np.linalg.solve(aMatrix, bVector)
```

Comment:
This all seems like a pretty convoluted way to figure out this 2x2 matrix.  It would be helpful for anyone coming along later trying to debug something if you could explain the steps involved here.  Maybe some mathy comments with what the matrix equation is that you are setting up and solving here.